app/assets/javascripts/discourse-common/lib/changes-dom.js.es6

```diff
@@ -0,0 +1,14 @@
+import wrap from "discourse-common/lib/wrap";
+
+export function DOMNotReadyError() { }
+DOMNotReadyError.prototype = new Error();
+
+export default function changesDOM(target, key, descriptor) {
+  const checkForDOM = function() {
+    if (!this.element || this.isDestroying || this.isDestroyed) {
+      throw new DOMNotReadyError();
+    }
+  };
+
+  return wrap(checkForDOM, DOMNotReadyError).call(this, target, key, descriptor);
```

Comment:
I spoke to @jjaffeux offline, but I we should replace the exception handling here.

Personally I think we could leave out `wrap` for now, but if you'd like to keep it we should return objects for the program flow rather than throw/catch exceptions.